### PR TITLE
nixos/nginx: Enable TLS 1.3 support

### PR DESCRIPTION
**Pull request by jtojnar**
###### Things done

- [x] Tested using sandboxing ([nix.useSandbox](http://nixos.org/nixos/manual/options.html#opt-nix.useSandbox) on NixOS, or option `sandbox` in [`nix.conf`](http://nixos.org/nix/manual/#sec-conf-file) on non-NixOS)
- Built on platform(s)
   - [x] NixOS
   - [ ] macOS
   - [ ] other Linux distributions
- [ ] Tested via one or more NixOS test(s) if existing and applicable for the change (look inside [nixos/tests](https://github.com/NixOS/nixpkgs/blob/master/nixos/tests))
- [ ] Tested compilation of all pkgs that depend on this change using `nix-shell -p nox --run "nox-review wip"`
- [ ] Tested execution of all binary files (usually in `./result/bin/`)
- [ ] Determined the impact on package closure size (by running `nix path-info -S` before and after)
- [ ] Assured whether relevant documentation is up to date
- [x] Fits [CONTRIBUTING.md](https://github.com/NixOS/nixpkgs/blob/master/.github/CONTRIBUTING.md).

---

### CODE REVIEW

File: nixos/doc/manual/release-notes/rl-1903.xml [hunk 1]
```diff
@@ -645,6 +645,9 @@
        This may break some older applications that still rely on those symbols.
        An upgrade guide can be found <link xlink:href="https://www.open-mpi.org/faq/?category=mpi-removed">here</link>.
      </para>
+    <para>
+     The nginx now relies on OpenSSL 1.1 and supports TLS 1.3 by default. You can change the supported protocols using <xref linkend="opt-services.nginx.sslProtocols"/>
```
Comment (thoughtpolice) [inline nixos/doc/manual/release-notes/rl-1903.xml]: Wordsmithing: "The nginx package now requires OpenSSL, and supports TLS 1.3 by default."

Comment (jtojnar) [reply nixos/doc/manual/release-notes/rl-1903.xml]: It always required OpenSSL, we just switched from 1.0 to 1.1 branch.

**Review by bachp — APPROVED** [review]

**Comment by dtzWill** [comment]
LGTM, test deployed to a server and works as expected! Thanks! :)

File: nixos/doc/manual/release-notes/rl-1903.xml (hunk 1 above)
Comment (thoughtpolice) [reply nixos/doc/manual/release-notes/rl-1903.xml]: I was more referring to the lack of the word "package" to qualify "nginx" to help it read easier, and placing emphasis on the "requirement" part. Yes, the full text should be:

> The nginx package now requires OpenSSL 1.1, and supports TLS 1.3 by default.

**Comment by lheckemann** [comment]
I've taken the liberty of rephrasing the release notes documentation myself so we can get this in today before the branch-off. A quick review would be appreciated!

**Comment by jtojnar** [comment]
Do not see any changes https://github.com/NixOS/nixpkgs/compare/3cedb8cbbf46b61bdfb9b0d138661c61eb057307..b3fbb0498a5a36c3e8082b4e73fc279acbde849a

**Comment by lheckemann** [comment]
Argh, my mistake. New attempt here.

**Review by andir — APPROVED** [review]
Looks good to me. The rephrasing also seems good.

**Comment by andir** [comment]
@lheckemann can you (back)port this to `release-19.03`?

**Comment by lheckemann** [comment]
Done! Thanks for the pointer.

**Comment by herrwiese** [comment]
Uhm… did commit 051e852 (actually build nginx with `openssl_1_1`) just fall under the table in the `release-19.03` channel? »You're not going to space^W^W^Wenabling TLS 1.3 today« ;)

**Comment by lheckemann** [comment]
:fearful: oops. Good catch, fixed!

**Pull request merged**